samcli/lib/remote_invoke/lambda_invoke_executors.py

```diff
@@ -0,0 +1,147 @@
+"""
+Remote invoke executor implementation for Lambda
+"""
+import base64
+import json
+import logging
+from json import JSONDecodeError
+from typing import Any, Dict, cast
+
+from botocore.exceptions import ClientError, ParamValidationError
+from botocore.response import StreamingBody
+
+from samcli.lib.remote_invoke.exceptions import InvalideBotoResponseException, InvalidResourceBotoParameterException
+from samcli.lib.remote_invoke.remote_invoke_executors import (
+    BotoActionExecutor,
+    RemoteInvokeExecutionInfo,
+    RemoteInvokeOutputFormat,
+    RemoteInvokeRequestResponseMapper,
+)
+from samcli.lib.utils import boto_utils
+
+LOG = logging.getLogger(__name__)
+FUNCTION_NAME = "FunctionName"
+PAYLOAD = "Payload"
+
+
+class LambdaInvokeExecutor(BotoActionExecutor):
+    """
+    Calls "invoke" method of "lambda" service with given input.
+    If a file location provided, the file handle will be passed as Payload object
+    """
+
+    _lambda_client: Any
+    _function_name: str
+
+    def __init__(self, lambda_client: Any, function_name: str):
+        self._lambda_client = lambda_client
+        self._function_name = function_name
+        self.request_parameters = {"InvocationType": "RequestResponse", "LogType": "Tail"}
+
+    def validate_action_parameters(self, parameters: dict) -> None:
+        """
+        Validates the input boto parameters and prepares the parameters for calling the API.
+
+        :param parameters: Boto parameters provided as input
+        """
+        for parameter_key, parameter_value in parameters.items():
+            if parameter_key == FUNCTION_NAME:
```

Comment:
can this ever expand apart from the two terms? we don't want this to become multiple if, elifs.

Reply:
Other than those 2 parameters, the additional parameters are directly passed to the boto API without any changes. If we do decide to process any other parameter, it would be added in here. For instance, the `ClientContext` parameter is required to be base64-encoded string. If we want to encode this before passing to the API, that logic would be added here.